src/client/Client.cc

```diff
@@ -8675,10 +8675,6 @@ int Client::_write(Fh *f, int64_t offset, uint64_t size, const char *buf,
   // use/adjust fd pos?
   if (offset < 0) {
     lock_fh_pos(f);
-    /*
-     * FIXME: this is racy in that we may block _after_ this point waiting for caps, and size may
-     * change out from under us.
-     */
```

Comment:
Probably could flesh out the justification for removing this a bit. Basically in order to do the _lseek to SEEK_END, you need to fetch the file size. So _lseek does a _getattr with CEPH_STAT_CAP_SIZE (aka CEPH_CAP_FILE_SHARED). At that point the size is effectively frozen until the client returns that cap back to the MDS, but that can't happen until the client_lock mutex is released.